### PR TITLE
Record and store parameters name to function's signature.

### PR DESCRIPTION
**Pull request by longit644**
Record and store parameters name to function's signature.

### CODE REVIEW

File: types/types.go [hunk 1]
```diff
@@ -412,12 +412,11 @@ func (m Member) String() string {
...
 type Signature struct {
-	// TODO: store the parameter names, not just types.
-
 	// If a method of some type, this is the type it's a member of.
-	Receiver   *Type
-	Parameters []*Type
-	Results    []*Type
+	Receiver           *Type
+	Parameters         []*Type
+	ParametersWithName []Parameter
```
Comment (lavalamp) [inline types/types.go]: `NamedParameters`

What do you think about instead just doing `ParameterNames []string`? It is awkward (I hate parallel arrays), but doesn't duplicate information or leave people scratching their heads quite as much about why we made the interface what it is.

Comment (lavalamp) [reply types/types.go]: We can claim they're separate since if the parameters are unnamed, they must all be unnamed :)

Comment (longit644) [reply types/types.go]: Yes, It makes sense. Using `ParameterNames []string` isn't the best practice, but it makes the interface clear.

**Comment by longit644** [comment]
Hi all, do I need to update anything?

**Pull request closed without merging**